paasta_tools/tron_tools.py

```diff
@@ -0,0 +1,371 @@
+        try:
+            return self['default_paasta_cluster']
+        except KeyError:
+            raise TronNotConfigured('Could not find default PaaSTA cluster in system Tron config')
+
+    def get_url(self):
+        """:returns The URL for the Tron master's API"""
+        try:
+            return self['url']
+        except KeyError:
+            raise TronNotConfigured('Could not find URL of Tron master in system Tron config')
+
+
+def load_tron_config():
+    return TronConfig(load_system_paasta_config().get_tron_config())
+
+
+def get_tron_client():
+    return TronClient(load_tron_config().get_url())
+
+
+def compose_instance(job, action):
+    return '%s%s%s' % (job, SPACER, action)
+
+
+def decompose_instance(instance):
+    """Get (job_name, action_name) from an instance."""
+    decomposed = instance.split(SPACER)
+    if len(decomposed) != 2:
+        raise InvalidInstanceConfig('Invalid instance name: %s' % instance)
+    return (decomposed[0], decomposed[1])
+
+
+class TronActionConfig(InstanceConfig):
+    config_filename_prefix = 'tron'
+
+    def __init__(self, service, instance, config_dict, branch_dict, soa_dir=DEFAULT_SOA_DIR):
+        super(TronActionConfig, self).__init__(
+            cluster=config_dict.get('cluster'),
+            instance=instance,
+            service=service,
+            config_dict=config_dict,
+            branch_dict=branch_dict,
+            soa_dir=soa_dir,
+        )
+        self.job, self.action = decompose_instance(instance)
+
+    def get_job_name(self):
+        return self.job
+
+    def get_action_name(self):
+        return self.action
+
+    def get_deploy_group(self):
+        return self.config_dict.get('deploy_group', '')
+
+    def get_cmd(self):
+        return self.config_dict.get('command')
+
+    def get_executor(self):
+        executor = self.config_dict.get('executor', 'ssh')
+        return 'mesos' if executor == 'paasta' else executor
+
+    def get_node(self):
+        return self.config_dict.get('node')
+
+    def get_retries(self):
+        return self.config_dict.get('retries')
+
+    def get_requires(self):
+        return self.config_dict.get('requires')
+
+    def get_calculated_constraints(self):
+        """Combine all configured Mesos constraints."""
+        constraints = self.get_constraints()
+        if constraints is not None:
+            return constraints
+        else:
+            constraints = self.get_extra_constraints()
+            constraints.extend(
+                self.get_deploy_constraints(
+                    blacklist=self.get_deploy_blacklist(),
+                    whitelist=self.get_deploy_whitelist(),
+                    # Don't have configs for the paasta cluster
+                    system_deploy_blacklist=[],
+                    system_deploy_whitelist=None,
+                ),
+            )
+            constraints.extend(self.get_pool_constraints())
+            assert isinstance(constraints, list)
```

Comment:
whats this for?

Reply:
dunno, another thing I copied from chronos tools :P